packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -78,6 +95,75 @@ Future<Depfile> copyAssets(Environment environment, Directory outputDirectory) a
   return Depfile(inputs + assetBundle.additionalDependencies, outputs);
 }
 
+/// The path of the SkSL JSON bundle included in flutter_assets.
+const String kSkSLShaderBundlePath = 'io.flutter.shaders.json';
+
+/// Validate and process an SkSL asset bundle in a [DevFSContent].
+///
+/// Returns `null` if the bundle was not provided, otherwise attempts to
+/// validate the bundle.
+///
+/// Throws [Exception] if the bundle is invalid due to formatting issues.
+///
+/// If the current target platform is different than the platform constructed
+/// for the bundle, a warning will be printed.
+DevFSContent processSkSLBundle(String bundlePath, {
+  @required TargetPlatform targetPlatform,
+  @required FileSystem fileSystem,
+  @required Logger logger,
+  @required String engineVersion,
+}) {
+  if (bundlePath == null) {
+    return null;
+  }
+  // Step 1: check that file exists.
+  final File skSLBundleFile = fileSystem.file(bundlePath);
+  if (!skSLBundleFile.existsSync()) {
+    logger.printError('$bundlePath does not exist.');
+    throw Exception('SkSL bundle was invalid.');
+  }
+
+  // Step 2: validate top level bundle structure.
+  Map<String, Object> bundle;
+  try {
+    bundle = json.decode(skSLBundleFile.readAsStringSync())
+      as Map<String, Object>;
+  } on FormatException {
+    logger.printError('"$bundle" was not a JSON object.');
```

Comment:
Print some of the FormatException details to give the user a hint.

Reply:
Done